src/vivarium/framework/results/interface.py

```diff
@@ -289,11 +297,11 @@ def register_unstratified_observation(
         ValueError
             If any required callable arguments are missing.
         """
-        required_callables = {
+        required_callables: dict[str, Callable[..., pd.DataFrame]] = {
```

Comment:
Shouldn't this be `dict[str, ResultsUpdater | ResultsFormatter | ResultsGatherer]`?

Reply:
I tried that, as well as the following (the arguments can either be one DataFrame or two DataFrames)

`Callable[[pd.DataFrame, ...], pd.DataFrame]`
`Callable[[pd.DataFrame], pd.DataFrame] | Callable[[pd.DataFrame, pd.DataFrame], pd.DataFrame]`
`Callable[[pd.DataFrame | [pd.DataFrame, pd.DataFrame]], pd.DataFrame]`

but they all gave me errors like the following (this one is from Callable[[pd.DataFrame,...], pd.DataFrame]

Argument 2 to "_check_for_required_callables" of "ResultsInterface" has incompatible type "dict[str, Callable[[DataFrame, Any], DataFrame]]"; expected "dict[str, Callable[[str, DataFrame], DataFrame] | Callable[[DataFrame], DataFrame] | Callable[[DataFrame, DataFrame], DataFrame]]"  [arg-type]